src/back/spv/writer.rs

```diff
@@ -865,15 +858,21 @@ impl Writer {
 
         let instruction;
 
-        match ty.inner.clone() {
+        match ty.inner {
             crate::TypeInner::Scalar { kind, width } => {
                 instruction = self.write_scalar(id, kind, width);
```

Comment:
I think this should use `get_or_create_scalar`

Reply:
No that is not possible, `get_or_create_scalar` and `get_or_create_vector` only returns a Word/Id, not an instruction.